### PR TITLE
Add definitions for mParticle SDK

### PR DESCRIPTION
**Pull request by rmi22186**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by RyanCavanaugh — APPROVED** [review]

File: types/mparticle__web-sdk/package.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "private": true,
+    "license": "Apache-2.0"
```
Comment (RyanCavanaugh) [inline types/mparticle__web-sdk/package.json:3]: Please remove; we do not support different licenses in this field.

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

File: types/mparticle__web-sdk/package.json (hunk 1 above)
Comment (rmi22186) [reply types/mparticle__web-sdk/package.json:3]: Got it.  I'll just remove the `package.json`, not sure why this is even here.

Thanks for the prompt PRs btw, first time PR-ing into DefinitelyTyped and seems like a pretty well oiled machine :)

File: types/mparticle__web-sdk/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,813 @@
+// Type definitions for mParticle/web-sdk SDK 2.12
+// Project: https://github.com/mParticle/mparticle-web-sdk
+// Definitions by: Alex Sapountzis <https://github.com/asap>
```
Comment (peterblazejewicz) [inline types/mparticle__web-sdk/index.d.ts:3]: You're referencing other user, why is this?

File: types/mparticle__web-sdk/tslint.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules":
+        {
+            "npm-naming": [true,{"mode":"code","errors":[["NeedsExportEquals",false]]}],
+            "object-literal-shorthand": false }
```
Comment (peterblazejewicz) [inline types/mparticle__web-sdk/tslint.json:6]:
```suggestion
    "extends": "dtslint/dt.json"
```

File: types/mparticle__web-sdk/test/mparticle__web-sdk-instance-tests.ts [hunk 4]
```diff
@@ -0,0 +1,721 @@
...
+    verbose: (error) => {
+        console.log(error);
+    },
+};
+
+const config: mParticle.MPConfiguration = {
+    isDevelopmentMode: true,
+    identifyRequest: identifyRequest,
+    identityCallback: identityCallback,
+    dataPlan: dataPlan,
```
Comment (peterblazejewicz) [inline types/mparticle__web-sdk/test/mparticle__web-sdk-instance-tests.ts:102]:
```suggestion
    identifyRequest,
    identityCallback,
    dataPlan,
```

File: types/mparticle__web-sdk/test/mparticle__web-sdk-instance-tests.ts [hunk 5]
```diff
@@ -0,0 +1,721 @@
...
+    appVersion: '1.0.0',
+    appName: 'testAppName',
+    logLevel: 'warning',
+    logger: logger,
```
Comment (peterblazejewicz) [inline types/mparticle__web-sdk/test/mparticle__web-sdk-instance-tests.ts:106]:
```suggestion
    logger,
```

File: types/mparticle__web-sdk/test/mparticle__web-sdk-tests.ts [hunk 6]
```diff
@@ -0,0 +1,719 @@
...
+    verbose: (error) => {
+        console.log(error);
+    },
+};
+
+const config: mParticle.MPConfiguration = {
+    isDevelopmentMode: true,
+    identifyRequest: identifyRequest,
+    identityCallback: identityCallback,
+    dataPlan: dataPlan,
```
Comment (peterblazejewicz) [inline types/mparticle__web-sdk/test/mparticle__web-sdk-tests.ts:98]:
```suggestion
    identifyRequest,
    identityCallback,
    dataPlan,
```

File: types/mparticle__web-sdk/test/mparticle__web-sdk-tests.ts [hunk 7]
```diff
@@ -0,0 +1,719 @@
...
+    appVersion: '1.0.0',
+    appName: 'testAppName',
+    logLevel: 'warning',
+    logger: logger,
```
Comment (peterblazejewicz) [inline types/mparticle__web-sdk/test/mparticle__web-sdk-tests.ts:102]:
```suggestion
    logger,
```

File: types/mparticle__web-sdk/index.d.ts [hunk 8]
```diff
@@ -0,0 +1,813 @@
...
+//                 Robert Ing <https://github.com/rmi22186>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.6
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
```
Comment (peterblazejewicz) [inline types/mparticle__web-sdk/index.d.ts:9]:
```suggestion
```
AFAIK you don't need this

File: types/mparticle__web-sdk/index.d.ts (hunk 2 above)
Comment (rmi22186) [reply types/mparticle__web-sdk/index.d.ts:3]: Alex actually helped me develop this as a typescript project in a hackthon previously which I am finishing up as part of thie PR, so I wanted to give him credit.

File: types/mparticle__web-sdk/index.d.ts (hunk 8 above)
Comment (rmi22186) [reply types/mparticle__web-sdk/index.d.ts:9]: Looks like you're right.

**Comment by rmi22186** [comment]
thanks for the feedback. let me know how this looks now @peterblazejewicz or @RyanCavanaugh

File: types/mparticle__web-sdk/index.d.ts (hunk 2 above)
Comment (peterblazejewicz) [reply types/mparticle__web-sdk/index.d.ts:3]: that's OK, but probably he should be part of PR review, to ACK using Github's handle here.
That's not enforced on DT, but good netiq IMO

Comment (rmi22186) [reply types/mparticle__web-sdk/index.d.ts:3]: That makes sense.  Adding him now, as we went through an internal approval process before I posted this PR for Definitely Typed.

**Review by RyanCavanaugh — APPROVED** [review]

**Comment by rmi22186** [comment]
Ready to merge

**Comment by rmi22186** [comment]
thanks @peterblazejewicz & @RyanCavanaugh!

<3 typescript

File: types/mparticle__web-sdk/index.d.ts [hunk 9]
```diff
@@ -0,0 +1,811 @@
...
+    remove: (product: Product, logEventBoolean?: boolean) => void;
+    /**
+     *
+     * @deprecated Cart persistence in mParticle has been deprecated.
+     */
+    clear: () => void;
+}
+
+export interface Product {
+    name: string;
```
Comment (jaanauati) [inline types/mparticle__web-sdk/index.d.ts:564]: Hi @rmi2218 !, qq, shouldn't names here start with capital letter?, i.e: `Name: string`, etc 
https://docs.mparticle.com/developers/sdk/web/commerce-tracking/
image

Comment (rmi22186) [reply types/mparticle__web-sdk/index.d.ts:564]: @jaanauati  - this has been resolved in this pr - https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59832 and is now available as [version 2.15.1](https://www.npmjs.com/package/@types/mparticle__web-sdk) for @types/mparticle__web-sdk

**Pull request merged**